apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,433 @@
+                    hostDevice(rh?.remoteHostId),
+                    bottomPadding = bottomPadding
+                )
+            }
+
+            ContactsLayout(
+                contactActions = {},
+                contactTypes = listOf(ContactType.REMOVED),
+            )
+        }
+    }
+}
+
+@Composable
+private fun ContactActionsSection(contactActions: @Composable () -> Unit, rh: RemoteHostInfo?) {
+    contactActions()
+    Spacer(Modifier.height(DEFAULT_PADDING))
+
+    val archived = remember { contactChats(chatModel.chats, listOf(ContactType.REMOVED)) }
+
+    if (archived.isNotEmpty()) {
+        SectionView(padding = PaddingValues(bottom = DEFAULT_PADDING)) {
+            SectionItemView(
+                click = {
+                    ModalManager.center.showCustomModal { close -> DeletedContactsView(
+                        rh = rh,
+                        close = close)
+                    }
+                }
+            ) {
+                Icon(
+                    painterResource(MR.images.ic_folder_open),
+                    contentDescription = stringResource(MR.strings.chat_deleted),
+                    tint = MaterialTheme.colors.secondary,
+                )
+                TextIconSpaced(extraPadding = true)
+                Text(text = stringResource(MR.strings.chat_deleted), color = MaterialTheme.colors.onBackground)
+            }
+        }
+    }
+}
+
+@Composable
+private fun ContactsLayout(
+    contactActions: @Composable () -> Unit,
+    contactTypes: List<ContactType>,
+    contactListTitle: String? = null) {
+
+    SectionView {
+        ContactsList(
+            contactTypes = contactTypes,
+            contactActions = contactActions,
+            contactListTitle = contactListTitle
+        )
+    }
+}
+
+@Composable
+fun ContactsView(
+    contactActions: @Composable () -> Unit,
+    rh: RemoteHostInfo?
+) {
+    ContactsLayout(
+        contactActions = { ContactActionsSection(contactActions, rh) },
+        contactTypes = listOf(ContactType.CARD, ContactType.RECENT, ContactType.REQUEST),
+        contactListTitle = stringResource(MR.strings.contact_list_header_title).uppercase()
+    )
+}
+
+@Composable
+private fun ContactsSearchBar(
+    listState: LazyListState,
+    searchText: MutableState<TextFieldValue>,
+    searchShowingSimplexLink: MutableState<Boolean>,
+    searchChatFilteredBySimplexLink: MutableState<String?>,
+    focused: Boolean,
+    onFocusChanged: (hasFocus: Boolean) -> Unit
+) {
+    Row(verticalAlignment = Alignment.CenterVertically) {
+        val focusRequester = remember { FocusRequester() }
+        Icon(painterResource(MR.images.ic_search), null, Modifier.padding(horizontal = DEFAULT_PADDING_HALF), tint = MaterialTheme.colors.secondary)
+        SearchTextField(
+            Modifier.weight(1f).onFocusChanged { onFocusChanged(it.hasFocus) }.focusRequester(focusRequester),
+            placeholder = stringResource(MR.strings.search_or_paste_simplex_link),
```

Comment:
interesting - we don't show groups in this list, but if user pasts link to a known group perhaps we should show that single group?

currently it shows "No filtered contacts" and correctly doesn't offer to connect, but we could "surprise" here and allow to open group via known link.

is it a difficult change? if so, maybe better to do it separately from this pr

Reply:
actually it should probably do same as Scan / Paste link view was doing before - open that group chat and show alert "Group already exists" (the interaction is different from chat list where it's shown in list without alert).

Scan / Paste link interaction is also broken btw - it correctly shows alert "Group already exists", but doesn't open chat - instead it goes back one modal, and if you close it the group chat is opened.

Reply:
same for known contact (known by address) - interaction is similarly broken on Scan / Paste link